pkg/provider/infisical/api/api.go

```diff
@@ -0,0 +1,255 @@
+	if err != nil {
+		return nil, err
+	}
+
+	rawRes, err := a.do(req) //nolint:bodyclose // linters bug
+	if err != nil {
+		return nil, err
+	}
+
+	var res RevokeMachineIdentityAccessTokenResponse
+	err = ReadAndUnmarshal(rawRes, &res)
+	if err != nil {
+		return nil, fmt.Errorf(errJSONSecretUnmarshal, err)
+	}
+	return &res, nil
+}
+
+func (a *InfisicalClient) GetSecretsV3(data GetSecretsV3Request) (map[string]string, error) {
+	endpointURL := a.resolveEndpoint("api/v3/secrets/raw")
+
+	req, err := http.NewRequest(http.MethodGet, endpointURL, http.NoBody)
+	q := req.URL.Query()
+	q.Add("workspaceSlug", data.ProjectSlug)
+	q.Add("environment", data.EnvironmentSlug)
+	q.Add("secretPath", data.SecretPath)
+	q.Add("include_imports", "true")
+	q.Add("expandSecretReferences", "true")
+	req.URL.RawQuery = q.Encode()
+
+	metrics.ObserveAPICall(constants.ProviderName, "GetSecretsV3", err)
+	if err != nil {
+		return nil, err
+	}
+
+	rawRes, err := a.do(req) //nolint:bodyclose // linters bug
+	if err != nil {
+		return nil, err
+	}
+
+	var res GetSecretsV3Response
+	err = ReadAndUnmarshal(rawRes, &res)
+	if err != nil {
+		return nil, fmt.Errorf(errJSONSecretUnmarshal, err)
+	}
+
+	secrets := make(map[string]string)
+	for _, s := range res.ImportedSecrets {
+		for _, el := range s.Secrets {
+			secrets[el.SecretKey] = el.SecretValue
+		}
+	}
+	for _, el := range res.Secrets {
+		secrets[el.SecretKey] = el.SecretValue
+	}
+
+	return secrets, nil
+}
+
+func (a *InfisicalClient) GetSecretByKeyV3(data GetSecretByKeyV3Request) (string, error) {
+	endpointURL := a.resolveEndpoint(fmt.Sprintf("api/v3/secrets/raw/%s", data.SecretKey))
+
+	req, err := http.NewRequest(http.MethodGet, endpointURL, http.NoBody)
+	q := req.URL.Query()
+	q.Add("workspaceSlug", data.ProjectSlug)
+	q.Add("environment", data.EnvironmentSlug)
+	q.Add("secretPath", data.SecretPath)
+	q.Add("include_imports", "true")
+	req.URL.RawQuery = q.Encode()
+
+	metrics.ObserveAPICall(constants.ProviderName, "GetSecretByKeyV3", err)
+	if err != nil {
+		return "", err
+	}
+
+	rawRes, err := a.do(req) //nolint:bodyclose // linters bug
+	if err != nil {
+		return "", err
+	}
+	if rawRes.StatusCode == 400 {
+		var errRes InfisicalAPIErrorResponse
+		err = ReadAndUnmarshal(rawRes, &errRes)
+		if err != nil {
+			return "", fmt.Errorf(errJSONSecretUnmarshal, err)
+		}
+
+		if errRes.Message == "Secret not found" {
+			return "", esv1beta1.NoSecretError{}
+		}
+		return "", errors.New(errRes.Message)
+	}
+
+	var res GetSecretByKeyV3Response
+	err = ReadAndUnmarshal(rawRes, &res)
+	if err != nil {
+		return "", fmt.Errorf(errJSONSecretUnmarshal, err)
+	}
+
+	return res.Secret.SecretValue, nil
+}
+
+func MarhalReqBody(data any) (*bytes.Reader, error) {
+	body, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	return bytes.NewReader(body), nil
+}
+
+func ReadAndUnmarshal(resp *http.Response, target any) error {
+	var buf bytes.Buffer
+	defer func() {
+		if resp.Body != nil {
```

Comment:
Don't need to check this. In case if there is no body, HTTP creates a nil body reader for which the close just does nothing.